src/Raven.Client/Documents/Session/InMemoryDocumentSessionOperations.cs

```diff
@@ -1278,6 +1279,76 @@ public void RegisterMissingIncludes(BlittableJsonReaderArray results, BlittableJ
             }
         }
 
+        internal void RegisterCounters(BlittableJsonReaderArray counters, bool gotAll, string[] ids)
+        {
+            if (CountersByDocId == null)
+            {
+                CountersByDocId = new Dictionary<string, (bool GotAll, Dictionary<string, long?> Values)>(StringComparer.OrdinalIgnoreCase);
+            }
+
+            if (counters == null || counters.Length == 0)
+            {
+                if (gotAll == false || ids.Length == 0)
```

Comment:
I think you have `||` but you meant `&&` here? what are you trying to check here?

Reply:
No, this is intentional actually. 
If I have no counters in result, and gotAll is true, and ids is not empty, then I wanna remember that by setting GotAll=true in cache for all the requested documents.

Reply:
In this case, you can skip the `ids.Length` check, because the `foreach` already has this behavior.